### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by jakubgs**
Changes:
- Implemented previously noted error handling
- Also added `--ignoreOptional` flag
- Renamed `fetcher.Pool` to `fetcher.Fetcher`
- Renamed `fetcher.NewPool` to `fetcher.New`

### CODE REVIEW

File: finder/finder.go [hunk 1]
```diff
@@ -10,24 +10,28 @@ import (
 	"github.com/status-im/go-maven-resolver/pom"
 )
 
+type Options struct {
+	IgnoreScopes    []string /* list of dependency scopes to ignore */
+	IgnoreOptional  bool     /* if optional dependencies should be ignored */
+	RecursiveSearch bool     /* recursive dependency resolution switch */
+}
+
```
Comment (Samyoul) [inline finder/finder.go:18]: This is nice.

File: read.go [hunk 2]
```diff
@@ -6,15 +6,20 @@ import (
...
 		return nil, err
 	}
-	defer file.Close() // TODO unhandled error
+
+	defer func() {
+		cerr := file.Close()
+		if cerr == nil {
+			err = cerr
+		}
+	}()
```
Comment (Samyoul) [inline read.go]: In this code the error is never returned. Looks like a tpyo as your other deferred funcs write this like below: 

```suggestion
	defer func() {
		cerr := file.Close()
		if err == nil {
			err = cerr
		}
	}()
```

File: pom/metadata.go [hunk 3]
```diff
@@ -21,13 +21,17 @@ type Metadata struct {
...
 /* For reading Metadata from downloaded XML file. */
-func MetadataFromReader(reader io.ReadCloser) (*Metadata, error) {
-	defer reader.Close() // TODO unhandled error
+func MetadataFromReader(reader io.ReadCloser) (meta *Metadata, err error) {
+	defer func() {
+		cerr := reader.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
```
Comment (Samyoul) [inline pom/metadata.go:30]: I know you probably didn't enjoy adding this code, as it looks dirty, but Go doesn't have a good pattern for handling errors caused in or by deferred functions

**Review by Samyoul — APPROVED** [review]
Apart from the tpyo this looks good to me.

File: pom/metadata.go (hunk 3 above)
Comment (jakubgs) [reply pom/metadata.go:30]: Yeah, it's ugly as sin, but what can you do. Can't wait for `1.15`.

File: read.go (hunk 2 above)
Comment (jakubgs) [reply read.go]: Ah yes, it is a typo.

**Pull request merged**